firestore/integration_test_internal/src/firestore_test.cc

```diff
@@ -1499,6 +1504,129 @@ TEST_F(FirestoreIntegrationTest, ClearPersistenceWhileRunningFails) {
   EXPECT_EQ(await_clear_persistence.error(), Error::kErrorFailedPrecondition);
 }
 
+TEST_F(FirestoreIntegrationTest, LegacyCacheConfigForMemoryCacheWorks) {
+  auto* db = TestFirestore("legacy_memory_cache");
+  auto settings = db->settings();
+  WITH_DEPRECATED_API(settings.set_persistence_enabled(false));
+  db->set_settings(std::move(settings));
+
+  Await(db->Document("rooms/eros")
+            .Set(MapFieldValue{{"desc", FieldValue::String("eros")}}));
+
+  auto get_future = db->Document("rooms/eros").Get(Source::kCache);
+  const DocumentSnapshot* snapshot = Await(get_future);
+  EXPECT_EQ(get_future.status(), FutureStatus::kFutureStatusComplete);
+  EXPECT_FALSE(snapshot->is_valid());
+  snapshot->id();
+}
+
+TEST_F(FirestoreIntegrationTest, LegacyCacheConfigForPersistenceCacheWorks) {
+  auto* db = TestFirestore("legacy_persistent_cache");
+  auto settings = db->settings();
+  WITH_DEPRECATED_API(settings.set_persistence_enabled(true));
+  db->set_settings(std::move(settings));
+
+  Await(db->Document("rooms/eros")
+            .Set(MapFieldValue{{"desc", FieldValue::String("eros")}}));
+
+  auto get_future = db->Document("rooms/eros").Get(Source::kCache);
+  const DocumentSnapshot* snapshot = Await(get_future);
+  EXPECT_EQ(get_future.status(), FutureStatus::kFutureStatusComplete);
+  EXPECT_TRUE(snapshot->is_valid());
+  EXPECT_THAT(snapshot->GetData(),
+              ContainerEq(MapFieldValue{{"desc", FieldValue::String("eros")}}));
+}
+
+TEST_F(FirestoreIntegrationTest, NewCacheConfigForMemoryCacheWorks) {
+  auto* db = TestFirestore("new_memory_cache");
+  auto settings = db->settings();
+  settings.set_local_cache_settings(MemoryCacheSettings::Create());
+  db->set_settings(std::move(settings));
+
+  Await(db->Document("rooms/eros")
```

Comment:
Each test case has a lot of duplicated logic to verify that persistence is in effect or is _not_ in effect. Consider moving this logic into a helper function called something like VerifyPersistenceEnabled(bool) to perform the checks. It also has the benefit of increasing the readability the individual test cases.